### DISCUSSION TITLE
How do i change icon of a specific file, that was referenced in another .json file?

### DISCUSSION THREAD

**Discussion by amqndin** [microsoft/vscode-discussions#979] [Extension Development QnA] [answered]
i've never made an extension before. i created typescript base for it with `yo code` thingy and asked ai to make it for me. but you know ai kinda sucks at coding so i had to ask the real developers. hope you can help!

**Comment by olguzzar** [comment, marked as answer]
Hi @amawdiin, the good news is that there is a lot of documentation around [creating your first extension](https://code.visualstudio.com/api/get-started/your-first-extension)! Please review the documentation, try it out, and let us know if there are any specific questions that you have.

**Reply by amqndin** [reply]
I have the extension, I didn't know how to make dynamic changes to icon because it is not documented anywhere, but people do it in their extensions. I have already looked into extension that does it and kinda got it that you need to change the actual theme.json for it. so you can count the post as resolved.
